[Wee Heng] IP

Decaxical opened the pull request:
### **DukePlus**

> "For every minute spent organizing, an hour is earned." – Benjamin Franklin ([source])(https://orderyourlife.com/blogs/blog/28-amazing-quotes-that-will-inspire-you-to-get-organized)

DukePlus helps you organize your tasks in the form of a task list!
It is:

- Text-based
- Intuitive
- ~~Useless~~ EXTRAORDINARILY USEFUL

All you have to do to start using this amazing product is:
1. Download it from [here](https://github.com/Decaxical/ip/releases).
2. Double-click it.
3. Add ALL your task.
4. Kick back and relax as DukePro manages your tasks for you 😉

Best part is DukePro ~~costs as much as it's worth. I mean priceless~~ is **FREE**!

 Features:

- [ ] Managing tasks
- [ ] Search for relevant tasks
- [ ] Sort according to the deadline (Coming soon)

If you are a Java Programmer, you can use it to practice Java too. Here's the `main` method:

```    
public static void main(String[] args) {
        initialize();
        run();
}
```

Commit:
Add Gradle support

Decaxical committed:
implement echoes

Echoes the commands inputted by the user

Decaxical committed:
List ability

implemented the ability to add tasks to a list and to display it

Decaxical committed:
Added mark and unmark task functionality

Added a new class task to help with marking and unmarking

Decaxical committed:
Implement Todo, Event, Deadline functionality

made Todo, Event and Deadline extend from task

Decaxical committed:
Text-UI-Testing update

added input.txt and expected.txt and ensure test passed

Decaxical committed:
Implemented DukeExceptions

Implemented DukeInvalidArgumentException and DukeException. handles when todo has no description and invalid command

Decaxical committed:
Added delete commannd

Allows for deletion of tasks

hqhqhq1 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,153 @@
+            }
+            Task taskToMark = taskList.get(taskToMarkNumber - 1);
+            taskToMark.markAsDone();
+            System.out.println("Duly noted. The following task has been marked as done");
```
maybe can add a method displayMark for the print message as what you did earlier 👍

hqhqhq1 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,153 @@
+            }
+            Task taskToUnmark = taskList.get(taskToUnmarkNumber - 1);
+            taskToUnmark.markAsNotDone();
+            System.out.println("Very well. The following task has been marked as undone");
```
same as above displayUnmark

hqhqhq1 commented on src/main/java/duke/tasks/Task.java:
```diff
@@ -0,0 +1,24 @@
+
+public class Task {
+    String content;
+    boolean markedDone = false;
```
boolean var can be named as isDone

tonishka replied:
```suggestion
    boolean isMarkedDone = false;
```

tonishka commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,153 @@
+            System.out.printf("%d. %s \n", taskCounter, task.toString());
+            taskCounter++;
+        }
+    }
```
Perhaps you can use a TaskList object to separate out each user's task lists and bundle away all related methods like displaying/adding/removing to the list?

tonishka commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,153 @@
+public class Duke {
+
+    private static ArrayList<Task> taskList;
+    private static Scanner getUserInput;
```
getUserInput sounds like a method name. Perhaps consider changing it to 
```suggestion
    private static Scanner sc;
```

tonishka commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,153 @@
+        System.out.printf("You now have %d item(s) in your list\n", taskList.size());
+    }
+
+    public static String[] parseArguments(String[] arguments) throws DukeInvalidArgumentException {
+        if (arguments.length < 2) {
+            throw new DukeInvalidArgumentException("There appears to be invalid arguments");
+        }
+
+        return arguments[1].split(" /([Aa][Tt]|[Bb][Yy]) ", 2);
+    }
```
Maybe have some JavaDoc comments to explain non-trivial public methods?

tonishka commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,153 @@
+package duke;
+
+import duke.exceptions.DukeInvalidArgumentException;
+import duke.tasks.*;
```
Perhaps list all imported classes explicitly?

Decaxical committed:
Added Saving functionality

Duke is now able to save and load the tasklist across different sessions

Decaxical committed:
Update Duke to recognize dates

Deadline and Event tasks now take in LocalDateTime

Decaxical committed:
Merge branch 'branch-Level-8'

Decaxical committed:
Revert "Merge branch 'branch-Level-8'"

This reverts commit a2ce70c3c592418575205755fe644006cba4a8ad, reversing
changes made to 74a00d18f388ce0192fb907ee2bea0da80d9d1d4.

Decaxical committed:
Revert "Added Saving functionality"

This reverts commit 74a00d18f388ce0192fb907ee2bea0da80d9d1d4.

Decaxical committed:
Revert "Update Duke to recognize dates"

This reverts commit 738ca509d4efdfc7981a69aef90aac99a1ed6b5d.

Decaxical committed:
Merge branch 'branch-Level-7'

Decaxical committed:
Revert "Revert "Merge branch 'branch-Level-8'""

This reverts commit 3f0f9036486b19e76ad376f27f72aa743ad2ee82.

Decaxical committed:
Revert "Revert "Added Saving functionality""

This reverts commit e6993ef6949ef74e4e0ed77b29f91f6244de9682.

Decaxical committed:
Storage save and load datetime

Added functionality for storage to load tasks time as localdatetime

Decaxical committed:
Extracted TextUI, Parser, Tasklist and Command into separate classes

Extracted TextUI for interactions with user, Storage to load and save tasklist, Parser to parse inputs and Tasklist which contains tasklist operations

Decaxical committed:
Cleaned up imports and packages

Removed unnecessary imports and fixed formatting

Decaxical committed:
Added JUnit tests for parser and Task

Added JUnit test for Parser parseCommand() and all functions in Task.
Also removed redundant functions from TaskList

Decaxical committed:
Changed compile version from java 15 to java 11

Rectified mistake in Intellij compiler and fixed related bugs like text blocks.
fixed minor bugs like mark, unmark and delete allowing exceed of size. Improve UI

Decaxical committed:
minor UI improvement

Decaxical committed:
Added Java documentation

Added Java documentation and fixed a bug: delete command displaying wrong task being deleted.

Decaxical committed:
Updated to uphold Coding Standard

Deleted unnecessary lines. Shortened code. Renamed variables.

Decaxical committed:
Added parseSaveDateTime

Accommodate different time format in saveData

Decaxical committed:
Added FindCommand

Allows users to search for tasks based on content.
Added a toString() for TaskList and removed DisplayTaskList() from ListCommand

Decaxical committed:
Merge branch 'branch-A-JavaDoc'

Decaxical committed:
Merge branch 'branch-Level-9'

Decaxical committed:
Updated Javadoc

Javadoc for Find Command and related changes

Decaxical committed:
Merge branch 'add-gradle-support'

Decaxical committed:
Updated gradle

Changed mainClassName in build.gradle

Decaxical committed:
Add GUI prototype

use FXML. Added resources and images.

Decaxical committed:
Add Assertions

Lack of assertions

Certain assumptions at various points in the code needs to be checked

Use assert method in java to validate assumptions like input

Let's add relevant assertions

Decaxical committed:
Update Code Quality

Overlook certain aspects of code quality

Make code more readable and standardizing code formats

Extracting certain functions and adopt "drill down" function approach

Let's adhere to the code quality in the textbook

Decaxical committed:
Revert "Update Code Quality"

This reverts commit eb1d2409854cf260c57ad0fca72d6089036cc3ff.

Decaxical committed:
Revert "Update Code Quality"

This reverts commit eb1d2409854cf260c57ad0fca72d6089036cc3ff.

Decaxical committed:
Revert "Add Assertions"

This reverts commit 75e98b3eeb64cf73136143e7febd2d2e7927ce0a.

Decaxical committed:
Merge branch 'branch-A-Assertions' of https://github.com/Decaxical/ip into branch-A-Assertions

Decaxical committed:
Update Code Quality

Overlook certain aspects of code quality

Make code more readable and standardizing code formats

Extracting certain functions and adopt "drill down" function approach

Let's adhere to the code quality in the textbook

Decaxical committed:
Merge pull request #2 from Decaxical/branch-A-CodeQuality

Update Code Quality

Decaxical committed:
Merge pull request #3 from Decaxical/branch-A-Assertions

Add Assertions

Decaxical committed:
Added Sort Functionality

Duke now has a sort command

sort command displays tasks based on what type of task they are and in order of when they are due. Task happening sooner are displayed first.

Decaxical committed:
Minor update for SortCommand

Update so Duke notifies user if a certain type of event is not in the list

Decaxical committed:
Updated User Guide

Added userguide on how to quick start, features and command summary

Decaxical committed:
Set theme jekyll-theme-slate

Decaxical committed:
Set theme jekyll-theme-slate

Decaxical committed:
Set theme jekyll-theme-minimal

Decaxical committed:
Update README.md

Decaxical committed:
Set theme jekyll-theme-midnight

Decaxical committed:
Update User Guide

Decaxical committed:
Update User Guide

Decaxical committed:
Update User Guide format

Decaxical committed:
Improved GUI

Added bubble message. Change image. Reduced window size.

Decaxical committed:
Added UI.png

Decaxical committed:
Minor update to User Guide format

Removed Data

Decaxical committed:
Minor bug fix update

parseindex and parseDatetime throws error if not parsable

Decaxical committed:
Update User Guide
